src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -55,168 +51,140 @@ public class ResolveFrameworkReferences : TaskBase
         [Output]
         public ITaskItem[] RuntimePacks { get; set; }
 
-        //  There should only be one AppHost item, but we use an item here so we can attach metadata to it
-        //  (ie the apphost pack name and version, and the relative path to the apphost inside of it so
-        //  we can resolve the full path later)
-        [Output]
-        public ITaskItem[] AppHost { get; set; }
-
+        //  Runtime packs which aren't available for the specified RuntimeIdentifier
         [Output]
-        public ITaskItem[] PackAsToolShimAppHosts { get; set; }
-
-        [Output]
-        public string[] UnresolvedFrameworkReferences { get; set; }
+        public ITaskItem[] UnavailableRuntimePacks { get; set; }
 
         protected override void ExecuteCore()
         {
-            var knownFrameworkReferences = KnownFrameworkReferences.Select(item => new KnownFrameworkReference(item))
+            //  Perf optimization: If there are no FrameworkReference items, then don't do anything
+            //  (This means that if you don't have any direct framework references, you won't get any transitive ones either
```

Comment:
I'm not sure this needs to be true. I would expect the transitive FrameworkReferences to be raised from assets files before we get here. Is it worth stating presumed implications on a feature that doesn't exist yet?

Reply:
This target runs before the assets file is read (it's the one that generates the list of packages that need to be downloaded).

Reply:
Ah, of course. The task name is a little ambiguous. I think of ResolveXReferences as something that turns abstract references into paths. Maybe we should rename it.

Reply:
I think the actual limitation in parentheses here is fine. I understand why it's needed now.